forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -220,6 +220,9 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
         if (fromBattlefield && !toBattlefield) {
             c.getController().setRevolt(true);
         }
+        if (zoneTo.is(ZoneType.Graveyard) && c.isPermanent()) {
+            c.getOwner().descend();
```

Comment:
this is way too early for this, the card might still be replaced to move somewhere else